### PR TITLE
Fix cmake warning under external flow

### LINKED ISSUE #3544: CMake warning when EXTERNAL_PROPRIETARY flag is used
### Bug Report
CMake warning is shown during SDL build when EXTERNAL_PROPRIETARY flag is used

##### Reproduction Steps
1. Build SDL in `EXTERNAL_PROPRIETARY` policy mode:
```
cmake ../sdl_core -DEXTENDED_POLICY=EXTERNAL_PROPRIETARY
```
##### Expected Behavior
No warning messages are displayed.

##### Observed Behavior
Warning message is shown:
```
CMake Warning (dev) at CMakeLists.txt:436 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "copy_policy_library" of target "smartDeviceLinkCore"
  does not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/release/7.0.0 (3ec2b6b)
* Testing Against: N/A

##### Notes
Issue is not applicable for `PROPRIETARY` and `HTTP` policy modes

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixes #3544 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Covered by CMake itself

### Summary
When CMake is called ../sdl_core -DEXTENDED_POLICY = EXTERNAL_PROPRIETARY we get a warning. Сmake did not find an added dependency. Different dependencies need to be added for different policies. EXTERNAL_PROPRIETARY has dependency "copy_library_Policy" and PROPRIETARY has copy_policy_library. So I added a policy check to establish the correct dependencies.
There was also an error with the lack of a new version of the CMP0054 policy. Added a new version of this policy.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by AKalinich-Luxoft** [comment]
@theresalech this PR is ready for a Livio review

File: CMakeLists.txt [hunk 1]
```diff
@@ -510,7 +511,11 @@ add_subdirectory(./src/appMain)
 add_subdirectory(./src/plugins)
 
 add_dependencies(${PROJECT} Policy)
-add_dependencies(${PROJECT} copy_policy_library)
+if(EXTENDED_POLICY STREQUAL "EXTERNAL_PROPRIETARY")
+  add_dependencies(${PROJECT} copy_library_Policy)
+else()
+  add_dependencies(${PROJECT} copy_policy_library)
+endif()
```
Comment (jacobkeeler) [inline CMakeLists.txt:518]: Would it not be easier to normalize the names of these targets rather than add this conditional? I say we just use the name `copy_policy_library` in both

**Review by jacobkeeler — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (AKalinich-Luxoft) [reply CMakeLists.txt]: @jacobkeeler good idea, thank you. Fixed in https://github.com/smartdevicelink/sdl_core/pull/3730/commits/f92ec7360f0dedbdbdc1ae726f51f1cb09dbbdcf

File: CMakeLists.txt [hunk 2]
```diff
@@ -29,7 +29,6 @@
 # POSSIBILITY OF SUCH DAMAGE.
 
 cmake_minimum_required(VERSION 2.8.11)
-cmake_policy(SET CMP0054 NEW)
```
Comment (jacobkeeler) [inline CMakeLists.txt]: This was separate from the comment I had. Is it planned to be included in a separate PR?

Comment (AKalinich-Luxoft) [reply CMakeLists.txt]: @jacobkeeler it was reverted by mistake. Restored in https://github.com/smartdevicelink/sdl_core/pull/3730/commits/763b1ef84c8e2dea7c3bcc9ee22b2b31fb3f91a9

**Review by jacobkeeler — APPROVED** [review]
